### PR TITLE
net-im/dino: add 0.4.0

### PR DESCRIPTION
**Pull request by jubalh**

### CODE REVIEW

File: net-im/dino/dino-0.4.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	dev-libs/icu
+	dev-libs/libgee:0.8
+	>=gui-libs/libadwaita-1.2.0:1 
+	net-libs/glib-networking
+	>=net-libs/libnice-0.1.15
+	net-libs/libsignal-protocol-c
+	net-libs/libsrtp:2
+	x11-libs/cairo
+	x11-libs/gdk-pixbuf:2
+	gui-libs/gtk:4
```
Comment (a17r) [inline net-im/dino/dino-0.4.0.ebuild:38]: That looks like it needs some new sorting after the category change

File: net-im/dino/dino-0.4.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	x11-libs/pango
+	gpg? ( app-crypt/gpgme:= )
+	http? ( net-libs/libsoup:2.4 )
+	omemo? (
+		dev-libs/libgcrypt:0
+		media-gfx/qrencode
+	)
+	notification-sound? ( media-libs/libcanberra:0[sound] )
```
Comment (a17r) [inline net-im/dino/dino-0.4.0.ebuild:46]: Line should be before omemo (per alphanumerical sorting)

File: net-im/dino/dino-0.4.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+	local disabled_plugins=(
+		$(usex gpg "" "openpgp")
+		$(usex omemo "" "omemo")
+		$(usex http  "" "http-files")
+	)
+	local enabled_plugins=(
+		$(usex notification-sound "notification-sound" "")
+	)
+	local mycmakeargs+=(
```
Comment (a17r) [inline net-im/dino/dino-0.4.0.ebuild:69]:
```suggestion
	local mycmakeargs=(
```

**Review by a17r — CHANGES_REQUESTED** [review]

File: net-im/dino/dino-0.4.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+fi
+
+RDEPEND="
+	app-text/gspell[vala]
+	dev-db/sqlite:3
+	dev-libs/glib:2
+	dev-libs/icu
+	dev-libs/libgee:0.8
+	gui-libs/gtk:4
+	>=gui-libs/libadwaita-1.2.0:1
```
Comment (a17r) [inline net-im/dino/dino-0.4.0.ebuild:32]: spotted a trailing space there

File: net-im/dino/dino-9999.ebuild [hunk 5]
```diff
@@ -28,13 +28,14 @@ RDEPEND="
...
 	dev-libs/libgee:0.8
+	>=gui-libs/libadwaita-1.2.0:1 
 	net-libs/glib-networking
 	>=net-libs/libnice-0.1.15
 	net-libs/libsignal-protocol-c
 	net-libs/libsrtp:2
 	x11-libs/cairo
 	x11-libs/gdk-pixbuf:2
-	x11-libs/gtk+:3
+	gui-libs/gtk:4
```
Comment (a17r) [inline net-im/dino/dino-9999.ebuild:38]: forward 0.4.0 review here too please ;)

File: net-im/dino/dino-0.4.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+else
+	KEYWORDS="~amd64 ~arm64"
+	SRC_URI="${MY_REPO_URI}/releases/download/v${PV}/${P}.tar.gz"
+fi
+
+RDEPEND="
+	app-text/gspell[vala]
+	dev-db/sqlite:3
+	dev-libs/glib:2
+	dev-libs/icu
```
Comment (a17r) [inline net-im/dino/dino-0.4.0.ebuild:29]: requires `:=` slot operator like every other icu revdep to trigger rebuild on ICU slot bumps.

File: net-im/dino/dino-0.4.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+	net-libs/libsrtp:2
+	x11-libs/cairo
+	x11-libs/gdk-pixbuf:2
+	x11-libs/pango
+	gpg? ( app-crypt/gpgme:= )
+	http? ( net-libs/libsoup:2.4 )
+	notification-sound? ( media-libs/libcanberra:0[sound] )
+	omemo? (
+		dev-libs/libgcrypt:0
+		media-gfx/qrencode
```
Comment (a17r) [inline net-im/dino/dino-0.4.0.ebuild:45]: missing `:=` slot operator on both deps here too

**Review by a17r — CHANGES_REQUESTED** [review]
Now that I'm not reviewing on my mobile:

``-- Could NOT find WebRTCAudioProcessing (missing: WebRTCAudioProcessing_LIBRARY) (Required is at least version "0.2")``

Do we have an automagic dependency here?

**Comment by jubalh** [comment]
> Do we have an automagic dependency here?

I decided to disable this for now since I don't use videocalls myself and don't know much about them.
If anybody else wants this I think they need to add the webrtc useflag, enable the `rtp` build option and add the dependencies mentioned in https://github.com/dino/dino/blob/master/plugins/rtp/CMakeLists.txt

**Comment by jubalh** [comment]
Is this good to go?

**Comment by a17r** [comment]
Forgot to mention that it makes sense to search on bugzilla for already reported bugs, I would add this to the commit summary:

Bug: https://bugs.gentoo.org/851765
Closes: https://bugs.gentoo.org/869971

Not sure what to make of that, and why gst-plugins-base is only in DEPEND:
https://bugs.gentoo.org/893854

**Comment by jubalh** [comment]
Ok mentioned the two bugs :)

**Pull request closed without merging**